pulsar/internal/topic_name.go

```diff
@@ -107,6 +107,9 @@ func ParseTopicName(topic string) (*TopicName, error) {
 }
 
 func TopicNameWithoutPartitionPart(tn *TopicName) string {
+	if tn == nil {
+		return ""
+	}
```

Comment:
A little confused here, under what circumstances, tn==nil here?

Reply:
![image](https://user-images.githubusercontent.com/20965307/163097738-10134c2f-7575-461d-9aef-e29b2f41fe6f.png)


Rather than dealing with the tn ==nil case here, I prefer here that we don't ignore err, or check both

Reply:
Checking both looks good to me.